### PR TITLE
ci: deduplicate code into test-core.sh

### PR DESCRIPTION
**Pull request by jgehrcke**
For addressing https://github.com/opstrace/opstrace/issues/1051.

### CODE REVIEW

**Comment by sreis** [comment]
I manually [started](https://buildkite.com/opstrace/upgrade-tests/builds/990) with `jp/sync-test-core` as the branch.

File: ci/test-core.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# - the name of that tenant
+# - the path to the authentication token file
+set -x
+set +e; RNDSTRING=$( tr -dc a-z < /dev/urandom | head -c 6 ); set -e
+TENANT_RND_NAME_FOR_TESTING_ADD_TENANT="testtenant${RNDSTRING}"
+./build/bin/opstrace ta-create-keypair ./ta-custom-keypair.pem
+./build/bin/opstrace ta-create-token "${OPSTRACE_CLUSTER_NAME}" \
+    "${TENANT_RND_NAME_FOR_TESTING_ADD_TENANT}" ta-custom-keypair.pem > tenant-rnd-auth-token-from-custom-keypair
+TENANT_RND_AUTHTOKEN="$(cat tenant-rnd-auth-token-from-custom-keypair)"
+./build/bin/opstrace ta-pubkeys-add \
```
Comment (sreis) [inline ci/test-core.sh:19]: The upgrade tests don't build the CLI so this [./build/bin/opstrace binary doesn't exist](https://buildkite.com/opstrace/upgrade-tests/builds/990#4dc318cf-b284-4ec9-aa2f-425a0df75f5f/6597-6602) it should point at `OPSTRACE_CLI_VERSION_TO`.

Comment (jgehrcke) [reply ci/test-core.sh:19]: ouch. Yes. Tomorrow. Or, feel free to add to this branch.

**Comment by sreis** [comment]
Add a commit to handle the path to the opstrace binary and also triggered the [upgrade tests](https://buildkite.com/opstrace/upgrade-tests/builds/992).

**Comment by sreis** [comment]
Upgrade tests ran but hit other issues. I'll track them tomorrow in a separate pr.

**Pull request merged**